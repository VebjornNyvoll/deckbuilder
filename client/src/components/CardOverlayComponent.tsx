import { useRef } from 'react';
import { Toast } from 'primereact/toast';
import { Column } from 'primereact/column';
import { DataTable } from 'primereact/datatable';
import { Button } from 'primereact/button';
import { gql, useMutation, useQuery } from '@apollo/client';
import { useContext } from 'react';
import { AuthContext } from '../context/authContext';

export const CardOverlayComponent = ({ cardId }) => {
  const context = useContext(AuthContext);

  const CARD_TO_DECK = gql`
    mutation AddCards($cardIds: [ID!]!, $deckId: ID!) {
      addCards(cardIds: $cardIds, deckId: $deckId) {
        error {
          error
          message
        }
      }
    }
  `;

  const GET_DECKS = gql`
    query Decks {
      user {
        decks {
          deckName
          id
        }
      }
    }
  `;

  const { loading, error, data } = useQuery(GET_DECKS);
  const decks =
    data?.user?.decks?.map((deck) => ({
      deckId: deck.id,
      name: deck.deckName,
    })) || [];

  const toast = useRef(null);

  const showMessage = (isError, content) => {
    toast.current.replace({
      severity: isError ? 'error' : 'success',
      summary: isError ? 'Error' : 'Success',
      detail: content,
      life: 3000,
    });
  };

  const [addCardsToDeck] = useMutation(CARD_TO_DECK);

  const addButtonTemplate = (rowData) => (
    <Button label="Add" className="p-button-sm" onClick={(e) => handleAddButtonClick(e, rowData)} />
  );

  const handleAddButtonClick = async (e, rowData) => {
    e.stopPropagation();
    try {
      const response = await addCardsToDeck({
        variables: {
          cardIds: [cardId],
          deckId: rowData.deckId,
        },
        context: {
          headers: {
            Authorization: `${context?.token}`,
          },
        },
      });

      if (response?.data?.addCards?.error?.error) {
        showMessage(true, `Could not add card to ${rowData.name}`);
      } else {
        showMessage(false, `Added card to ${rowData.name}`);
      }
    } catch (error) {
      showMessage(true, `An error occurred: ${error.message}`);
    }
  };

  if (loading) return <p data-testid={cardId}>Loading decks...</p>;
  if (error) return <p>Error loading decks</p>;

  const nameBodyTemplate = (rowData) => <span style={{ cursor: 'pointer' }}>{rowData.name}</span>;

  return (
    <div
      className="card flex flex-column align-items-center gap-3"
      style={{
        width: '100%',
        maxWidth: '30rem',
        overflowY: 'auto',
        maxHeight: '400px',
      }}
    >
      <Toast ref={toast} />
      <DataTable sortOrder={-1} sortField={'name'} value={decks} tableStyle={{ width: '100%', maxWidth: '30rem' }}>
        <Column sortable field="name" header="Deck" body={nameBodyTemplate}></Column>
        <Column body={addButtonTemplate}></Column>
      </DataTable>
    </div>
  );
};
