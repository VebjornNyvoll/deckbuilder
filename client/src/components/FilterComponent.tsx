import React, { useState } from 'react';
import { SelectButton } from 'primereact/selectbutton';
import { InputText } from 'primereact/inputtext';
import { Dropdown } from 'primereact/dropdown';
import "../Filters.css";
import filters from "../data/filters.json";

export default function FilterComponent() {
    const [selectedFilter, setSelectedFilter] = useState(null);
    const [filterValue, setFilterValue] = useState('');
    const [filterInput, setFilterInput] = useState(null);
    const [comparison, setComparison] = useState(null);
    const [sorting, setSorting] = useState(null);
    const sortOptions = [
        { label: 'Ascending', value: 'asc' },
        { label: 'Descending', value: 'desc' }
    ];

    const comparisonOptions = [
        { label: 'Greater than', value: 'gt' },
        { label: 'Less than', value: 'lt' }
    ];

    const filterItems = filters.map(filter => ({
        label: filter.keyName,
        value: filter,
    }));

    const onFilterChange = (e) => {
        const filterData = e.value;
        setSelectedFilter(filterData); // Save the entire filter object

        // Reset filter input and value
        setFilterInput(null);
        setFilterValue('');
        setComparison(null);
        setSorting(null);

        switch (filterData.type) {
            case 'string':
                setFilterInput('string');
                break;
            case 'integer':
                setFilterInput('integer');
                break;
            case 'boolean':
                setFilterInput('boolean');
                break;
            default:
                setFilterInput(null);
                break;
        }
    };

    const renderFilterInput = () => {
        switch (filterInput) {
            case 'string':
                return (
                    <div>

                   
                    <>
                    <InputText
                        value={filterValue}
                        onChange={(e) => setFilterValue(e.target.value)}
                        placeholder="Search"
                    />
                     <Dropdown
                            value={sorting}
                            options={sortOptions}
                            onChange={(e) => setSorting(e.value)}
                            placeholder="Select Sorting"
                        />
                    </>
                    </div>
                    

                    
                );
            case 'integer':
                return (
                    <div>
                        
                    <>
                        <InputText
                            keyfilter="int"
                            value={filterValue}
                            onChange={(e) => setFilterValue(e.target.value)}
                            placeholder="Enter number"
                        />
                        <Dropdown
                            value={comparison}
                            options={comparisonOptions}
                            onChange={(e) => setComparison(e.value)}
                            placeholder="Select Comparison"
                        />
                        <Dropdown
                            value={sorting}
                            options={sortOptions}
                            onChange={(e) => setSorting(e.value)}
                            placeholder="Select Sorting"
                        />
                    </>
                    
                    </div>
                );
                case 'boolean':
                    return (
                        <div>
                        <>
                            <InputText
                                keyfilter="int"
                                value={filterValue}
                                onChange={(e) => setFilterValue(e.target.value)}
                                placeholder="Enter number"
                            />
                            <Dropdown
                                value={comparison}
                                options={comparisonOptions}
                                onChange={(e) => setComparison(e.value)}
                                placeholder="Select Comparison"
                            />
                            <Dropdown
                                value={sorting}
                                options={sortOptions}
                                onChange={(e) => setSorting(e.value)}
                                placeholder="Select Sorting"
                            />
                        </>
                        </div>
                    );
            
            default:
                return null;
        }
    };

    return (
        <div className="card flex justify-content-center attributes">
            <SelectButton
                value={selectedFilter}
                onChange={onFilterChange}
                optionLabel="label"
                options={filterItems}
            />
            {renderFilterInput()}
        </div>
    );
}
