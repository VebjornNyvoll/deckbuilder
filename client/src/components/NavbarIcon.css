i {
  margin-right: 8px;
}
