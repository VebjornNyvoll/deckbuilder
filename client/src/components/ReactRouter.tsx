import { Route, Routes } from 'react-router-dom';
import DetailedView from './DetailedView';
import Index from '../Index';
import Deck from '../Deck';
import Login from '../Login';
import CreateAccount from '../CreateAccount';

const ReactRouter: React.FC = () => {
    return (
        <Routes>
            <Route path="/detail/:cardId" element={<DetailedView />} />
            <Route path="/login" element={<Login/>} />
            <Route path="/create-account" element={<CreateAccount/>}/>
            <Route index element={<Index />} />
            <Route path="/decks" element={<Deck/>}/>
        </Routes>
    );
};

export default ReactRouter;
