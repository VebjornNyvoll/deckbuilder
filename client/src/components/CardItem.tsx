import React, { useRef, useState, useEffect } from "react";
import { Tag } from "primereact/tag";
import { Button } from "primereact/button";
import CardTooltip from "./CardTooltip";
import { Dialog } from "primereact/dialog";
import {CardOverlayComponent} from "./CardOverlayComponent";
import { OverlayPanel } from "primereact/overlaypanel";
import { useAppSelector } from "../service/hooks";
import { CardService } from "../service/CardService";
import parse from "html-react-parser";

interface CardItemProps {
  card: Card;
  onClick?: (cardId: string) => void;
}
interface Mechanic {
  name: string;
}
export interface Card {
  id: string;
  cardId: string;
  dbfId: number;
  name: string;
  cardSet: string;
  type: string;
  faction: string;
  rarity: string;
  cost: number;
  attack: number;
  health: number;
  text: string;
  flavor: string;
  artist: string;
  collectible: boolean;
  elite: boolean;
  playerClass: string;
  img: string;
  imgGold: string | undefined;
  locale: string;
  mechanics: Mechanic[];
}
const getSeverity = (card: Card) => {
  switch (card.faction) {
    case "Alliance":
      return "success";
    case "Scourge":
      return "success";
    case "Explorer":
      return "warning";
    case "Legion":
      return "warning";
    case "Horde":
      return "danger";
    case "Empire":
      return "danger";
    case "Pirate":
      return "danger";
    case "Neutral":
      return "info";
    default:
      return null;
  }
};

export function CardPopUp(props: { cardId: string; open: boolean; onClose: () => void }) {
  const { open, onClose, cardId } = props;
  const [card, setCard] = useState(CardService.getEmptyCard());


  useEffect(() => {
    if (cardId) {
      CardService.getCardById(cardId)
        .then(setCard)
        .catch((error) => {
          console.error("Error fetching card:", error);
        });
    }
  }, [cardId]);

  const renderCardProperties = () => {
    if (!card) return <p>Loading card details...</p>;
  };

  const headerContent = <div style={{ textAlign: "center" }}>{card ? card.name : "Loading..."}</div>;

  return (
      <Dialog
          headerStyle={{ height: "70px" }}
          header={headerContent}
          visible={open}
          style={{ width: "60vw", maxWidth: "400px" }}
          onHide={onClose}
      >
        <div
           tabIndex={0} style={{ textAlign: "left" }}>
              {card && (
              <>
              {card.attack && <p>Attack: {card.attack}</p>}
        {card.health && <p>Health: {card.health}</p>}
          {card.cost != null && <p>Cost: {card.cost}</p>}
            {card.rarity && <p>Rarity: {card.rarity}</p>}
          {card.text && <p style={{ whiteSpace: "pre-wrap" }}>
            Text: {parse(card.text .replace(/\[x\]|#|\$|\\n/g, " "))}
          </p>}
          {card.flavor && <p>Flavor: {card.flavor}</p>}
            {card.faction && <p>Faction: {card.faction}</p>}
            {card.cardSet && <p>Cardset: {card.cardSet}</p>}
            {card.playerClass && <p>Player class: {card.playerClass}</p>}
            {card.artist && <p>Artist: {card.artist}</p>}
            <br />
            {renderCardProperties()}
          </>
        )}
        </div>
      </Dialog>
  );
}

export const ListItem: React.FC<CardItemProps> = ({ card, onClick }) => {
  const dataSaver = useAppSelector((state) => state.datasaver.datasaver);
  const idString = card.id.toString();
  const op = useRef(null);
  const showOverlayPanel = (event: React.SyntheticEvent<Element, Event>) => {
    // Prevent event from bubbling up to parent elements
    event.stopPropagation();
    if (op.current && (op.current as OverlayPanel).toggle) {
      (op.current as OverlayPanel).toggle(event);
    }
  };
  //The onClick as defined in Props must take in a argument card
  const handleItemClick = () => {
    if (onClick) {
      onClick(card.id);
    }
  };
  return (
    //Needs to send to const to do the onClick that has been sent.
    <button onClick={handleItemClick}  aria-haspopup aria-labelledby={idString+"set "+idString+"faction "+idString+"name"} className="col-12 border-1 surface-border surface-card border-round " >
      <div className="flex rem flex-row xl:align-items-start p-4 gap-4">
        {!dataSaver && (<img
          className="w-10rem shadow-2 block xl:block mx-auto border-round align-items-center"
          src={card.img}
          alt={card.name}
        />
        )}
        <div className="flex flex-column sm:flex-row justify-content-between align-items-center xl:align-items-start flex-1 gap-4">
          <div className="flex flex-column align-items-center sm:align-items-start gap-3">
            <div id={idString+"name"} className="text-2xl font-bold text-900">{card.name}</div>
            {/* Additional card details */}
            <div className="flex align-items-center gap-3">
              <span className="flex align-items-center gap-2">
                <i className="pi pi-book"></i>
                <span className="font-semibold">{card.cardSet}</span>
              </span>
              <Tag
                value={card.faction ? card.faction.toString() : "None"}
                severity={getSeverity(card)}
              ></Tag>
            </div>
            {dataSaver && (
            <div>
              <p className="p-0 m-0">
                {card.attack
                  ? "Attack: " + card.attack.toString()
                  : "No attack"}
              </p>
              <p>
                {card.health
                  ? "Health: " + card.health.toString()
                  : "No health"}
              </p>
            </div>
            )}
          </div>
          <div className="flex sm:flex-column align-items-center sm:align-items-end gap-3 sm:gap-2">
            { <span className="text-2xl font-semibold">
              {card.type ? card.type.toString() : "No Type"}
            </span> }
            <OverlayPanel ref={op} dismissable>
              <CardOverlayComponent op={op} cardId={card.id} />
            </OverlayPanel>
            <Button
              id = {idString+"btn"}
              title={"Add Card: "+ card.name.toString()}
              aria-labelledby={idString+"btn"}
              icon="pi pi-plus"
              className="p-button-rounded"
              onClick={showOverlayPanel}
            />
          </div>
        </div>
      </div>
    </button>
  );
};


export const GridItem: React.FC<CardItemProps> = ({ card, onClick }) => {
  const dataSaver = useAppSelector((state) => state.datasaver.datasaver);
  const idString = card.id.toString();
  const op = useRef(null);
  const showOverlayPanel = (event: React.SyntheticEvent<Element, Event>) => {
    // Prevent event from bubbling up to parent elements
    event.stopPropagation();
    if (op.current && (op.current as OverlayPanel).toggle) {
      (op.current as OverlayPanel).toggle(event);
    }
  };
  const handleItemClick = () => {
    if (onClick) {
      onClick(card.id);
    }
  };
  return (
    <div className="col-12 sm:col-6 lg:col-4 xl:col-3 p-2">
      <button style={{width: "100%"}} className="p-4 border-1 surface-border surface-card border-round" aria-haspopup aria-labelledby={idString+"set "+idString+"faction "+idString+"name"} role="button" onClick={handleItemClick}>
        <div>
          <div  className="flex align-items-center justify-content-between gap-2">
            <div className="flex align-items-center gap-2">
              <i className="pi pi-book"></i>{" "}
              <span id={idString+"set"} className="font-semibold text-xs">{card.cardSet}</span>
            </div>
            <Tag id={idString+"faction"}
              value={card.faction ? card.faction.toString() : "None"}
              severity={getSeverity(card)}
            ></Tag>
          </div>
          <div className="flex flex-column align-items-center gap-3 py-5">
          {!dataSaver && (
            <img
              className={"w-9 shadow-2 border-round"}
              src={card.img}
              alt={card.name}
            />
          )}

            <div className="text-l font-bold">{card.name}</div>
            {dataSaver && (
            <>
              <CardTooltip card={card}></CardTooltip>
              <p className="p-0 m-0 align-items-center">
                {card.attack ? "Attack: " + card.attack.toString() : "No attack"}
              </p>
              <p>
                {card.health ? "Health: " + card.health.toString() : "No health"}
              </p>
            </>
          )}
        </div>
          <div className="flex align-items-center justify-content-between">
            { <span className="text-2xl font-semibold">
              {card.type ? card.type.toString() : "No Type"}
            </span> }
            <Button
              id = {idString+"btn"}
              title={"Add Card: "+ card.name.toString()}
              aria-labelledby={idString+"btn"}
              icon="pi pi-plus"
              className="p-button-rounded"
              onClick={showOverlayPanel}
            />
          </div>
          <OverlayPanel ref={op} dismissable>
            <CardOverlayComponent op={op} cardId={card.id} />
          </OverlayPanel>
        </div>
      </button>
    </div>
  );
};
