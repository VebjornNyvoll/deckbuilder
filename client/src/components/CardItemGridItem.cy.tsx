import React from 'react'
import { GridItem } from './CardItem'

describe('<GridItem />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-react
    cy.mount(<GridItem />)
  })
})