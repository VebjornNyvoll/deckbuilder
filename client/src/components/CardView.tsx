import { useState, useEffect, useRef } from "react";
import { DataView } from "primereact/dataview";
import { ListItem, GridItem, Card, CardPopUp } from "./CardItem";
import { RemoveScroll } from "react-remove-scroll";
import { ScrollTop } from "primereact/scrolltop";
import { useAppDispatch, useAppSelector } from "../service/hooks";
import { setCards, addCards } from "../service/cards/cardsSlice";
import { CardService } from "../service/CardService";

export default function CardView() {
  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
  const [popCard, setPopCard] = useState<Card | undefined>();
  const [loading, setLoading] = useState(false);
  const [hasMore, setHasMore] = useState(true);
  const scrollContainerRef = useRef<HTMLDivElement | null>(null);
  const dispatch = useAppDispatch();
  const filters = useAppSelector((state) => state.filters);
  const sort = useAppSelector((state) => state.sort);
  const cards = useAppSelector((state) => state.cards.cards); // Access cards from Redux state
  const layout = useAppSelector((state) => state.layout.layout);

  const options = {
    limit: 20,
    skip: 0,
    sortBy: sort,
  };

  useEffect(() => {
    options.limit = 20; // Reset limit when filters change
    options.skip = 0; // Reset skip when filters change
    options.sortBy = sort; // Keep the sort options when filters change
    loadInitialCards();
  }, [filters, sort]);

  useEffect(() => {
    loadInitialCards();
  }, []);

  const loadInitialCards = () => {
    if (loading) return;
    setLoading(true);

    CardService.getFilteredCards(filters, options)
      .then((data) => {
        if (data.cards) {
          dispatch(setCards(data.cards)); // Dispatch the action to set initial cards
          setHasMore(true);
        } else {
          dispatch(setCards([])); // Dispatch the action to set an empty array
          setHasMore(false);
        }
        setLoading(false);
      })
      .catch((error) => {
        console.error("Error fetching cards:", error);
        dispatch(setCards([])); // Dispatch the action to set an empty array on error
        setHasMore(false);
        setLoading(false);
      });
  };

  const loadMoreCards = () => {
    if (loading || !hasMore) return;
    setLoading(true);
    options.skip = cards.length;

    CardService.getFilteredCards(filters, options)
      .then((data) => {
        if (data.cards && data.hasNextPage) {
          dispatch(addCards(data.cards)); // Dispatch the action to add more cards
          setHasMore(true);
        } else {
          setHasMore(false);
        }
        setLoading(false);
      })
      .catch((error) => {
        console.error("Error fetching more cards:", error);
        setHasMore(false);
        setLoading(false);
      });
  };

  const handleScroll = () => {
    if (loading || !hasMore) return;

    const scrollY = scrollContainerRef.current.scrollTop;
    const windowHeight = scrollContainerRef.current.clientHeight;
    const contentHeight = scrollContainerRef.current.scrollHeight;

    if (contentHeight - (scrollY + windowHeight) < 500) {
      loadMoreCards();
    }
  };

  useEffect(() => {
    if (scrollContainerRef.current) {
      scrollContainerRef.current.addEventListener("scroll", handleScroll);
    }

    return () => {
      if (scrollContainerRef.current) {
        scrollContainerRef.current.removeEventListener("scroll", handleScroll);
      }
    };
  }, [handleScroll, scrollContainerRef]);

  const openDialog = (card: Card) => {
    setPopCard(card);
    setIsDialogOpen(true);
  };

  const closeDialog = () => {
    setIsDialogOpen(false);
  };

  const itemTemplate = (card: Card, layout: string) => {
    const handleClick = () => {
      openDialog(card);
    };

    if (!card) {
      return null;
    }
    if (layout === "list") {
      return <ListItem card={card} onClick={handleClick} />;
    } else if (layout === "grid") {
      return <GridItem card={card} onClick={handleClick} />;
    }
  };

  return (
    <RemoveScroll>
      <div
        className="card"
        ref={scrollContainerRef}
        style={{ height: "calc(100vh - 62px)", overflow: "auto" }}
      >
        <DataView value={cards} itemTemplate={itemTemplate} layout={layout} />
        {popCard && (
          <CardPopUp card={popCard} open={isDialogOpen} onClose={closeDialog} />
        )}
        <ScrollTop
          target="parent"
          threshold={1000}
          className="w-3rem h-3rem border-round bg-primary"
          icon="pi pi-arrow-up text-base"
        />
      </div>
    </RemoveScroll>
  );
}
