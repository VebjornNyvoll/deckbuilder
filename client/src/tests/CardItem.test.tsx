import { describe, test, expect } from 'vitest';
import { ListItem } from '../components/CardItem';
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { ApolloProvider } from '@apollo/client';
import client from '../service/apolloClient';
import jsdom from 'jsdom';

const virtualConsole = new jsdom.VirtualConsole();
virtualConsole.on('error', () => {
  // No-op to skip console errors.
});

const initialState = {
  filters: {},
  sort: {},
  layout: {
    layout: 'grid',
  },
  datasaver: {
    datasaver: false,
  },

  cards: {
    cards: [],
  },
};
const mockStore = configureStore();
const store = mockStore(initialState);

interface Mechanic {
  name: string;
}
interface Card {
  id: string;
  cardId: string;
  dbfId: number;
  name: string;
  cardSet: string;
  type: string;
  faction: string;
  rarity: string;
  cost: number;
  attack: number;
  health: number;
  text: string;
  flavor: string;
  artist: string;
  collectible: boolean;
  elite: boolean;
  playerClass: string;
  img: string;
  imgGold: string | undefined;
  locale: string;
  mechanics: Mechanic[];
}

const mockupCard: Card = {
  //__typename: "Card",
  attack: 6,
  cardSet: 'Mean Streets of Gadgetzan',
  cost: 6,
  faction: '',
  health: 6,
  id: '655caba86ead681c6450b0e7',
  img: 'https://d15f34w2p8l1cc.cloudfront.net/hearthstone/f8b2f6eef5466fb1fb0bed9cc06c667e032df2b847bc6b72fb1e3bca28b4c939.png',
  name: '"Little Friend"',
  type: 'Minion',

  cardId: 'string',
  dbfId: 2,
  rarity: 'string',
  text: 'string',
  flavor: 'string',
  artist: 'string',
  collectible: true,
  elite: true,
  playerClass: 'string',
  imgGold: 'string',
  locale: 'string',
  mechanics: [],
};

describe('Card item test', async () => {
  test('popup test', async () => {
    render(
      <Provider store={store}>
        <ApolloProvider client={client}>
          <Router>
            <ListItem card={mockupCard}></ListItem>
          </Router>
        </ApolloProvider>
      </Provider>,
    );

    const overlayButton = screen.getByTestId('overlayPanelButton');
    if (overlayButton) {
      fireEvent.click(overlayButton);
    } else {
      console.log('Overlay button not found');
    }

    const panel = screen.getByTestId(mockupCard.id);
    expect(panel.innerHTML).toEqual('Loading decks...');
  });
});

describe('Card item matches snapshot', async () => {
  test('snapshot test', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ApolloProvider client={client}>
          <Router>
            <ListItem card={mockupCard}></ListItem>
          </Router>
        </ApolloProvider>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
