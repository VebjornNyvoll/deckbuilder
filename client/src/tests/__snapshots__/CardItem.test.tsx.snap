// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card item matches snapshot > snapshot test 1`] = `
<DocumentFragment>
  <div
    class="col-12 border-1 surface-border surface-card border-round "
  >
    <button
      aria-haspopup="true"
      aria-labelledby="stringset stringfaction stringname"
      class="col-12 border-1 surface-border surface-card border-round cursor-pointer"
    >
      <div
        class="flex rem flex-row xl:align-items-start p-4 gap-4"
      >
        <img
          alt=""
          class="w-10rem shadow-2 block xl:block mx-auto border-round align-items-center"
          friend=""
          little=""
          src="https://d15f34w2p8l1cc.cloudfront.net/hearthstone/f8b2f6eef5466fb1fb0bed9cc06c667e032df2b847bc6b72fb1e3bca28b4c939.png"
        />
        <div
          class="flex flex-column sm:flex-row justify-content-between align-items-center xl:align-items-start flex-1 gap-4"
        >
          <div
            class="flex flex-column align-items-center sm:align-items-start gap-3"
          >
            <div
              class="text-2xl font-bold text-900"
              id="stringname"
            >
              "Little Friend"
            </div>
            <div
              class="flex align-items-center gap-3"
            >
              <span
                class="flex align-items-center gap-2"
              >
                <i
                  class="pi pi-book"
                />
                <span
                  class="font-semibold"
                  id="stringset"
                >
                  Mean Streets of Gadgetzan
                </span>
              </span>
              <span
                class="p-tag p-component"
                data-pc-name="tag"
                data-pc-section="root"
                style=""
              >
                <span
                  class="p-tag-value"
                  data-pc-section="value"
                >
                  None
                </span>
                <span />
              </span>
            </div>
          </div>
          <div
            class="flex sm:flex-column align-items-center sm:align-items-end gap-3 sm:gap-2"
          >
            <span
              class="text-2xl font-semibold"
            >
              Minion
            </span>
          </div>
        </div>
      </div>
    </button>
    <button
      aria-labelledby="stringbtn"
      class="p-button-rounded p-button p-component p-button-icon-only"
      data-pc-name="button"
      data-pc-section="root"
      data-testid="overlayPanelButton"
      friend=""
      id="stringbtn"
      little=""
      title="Add Card: "
    >
      <span
        class="p-button-icon p-c pi pi-plus"
        data-pc-section="icon"
      />
      <span
        class="p-button-label p-c"
      >
        Â 
      </span>
    </button>
  </div>
</DocumentFragment>
`;
